app/models/test_track/fake/visitor.rb

```diff
@@ -21,21 +19,16 @@ def assignments
     @assignments ||= _assignments
   end
 
+  def split_registry
+    TestTrack::Fake::SplitRegistry.instance.to_h
+  end
+
   private
 
   def _assignments
-    TestTrack::Fake::SplitRegistry.instance.splits.map do |split|
-      index = hash_fixnum(split.name) % split.registry.keys.size
-      variant = split.registry.keys[index]
-      Assignment.new(split.name, variant, false, "the_context")
+    split_registry.keys.map do |split_name|
+      variant = TestTrack::VariantCalculator.new(visitor: self, split_name: split_name).variant
```

Comment:
I changed this to use the VariantCalculator rather than the partial implementation we had before. I think it cleans things up a bit, but its not strictly necessary.